src/factor_instances_provider/provider/provider_adopters/securify_entity_factor_instances_provider.rs

```diff
@@ -0,0 +1,244 @@
+    /// split into factor instance to use directly and factor instances which was cached, into
+    /// the mutable `cache` parameter.
+    ///
+    /// We are always reading from the beginning of each FactorInstance collection in the cache,
+    /// and we are always appending to the end.
+    pub async fn for_account_mfa(
+        cache: &mut FactorInstancesCache,
+        matrix_of_factor_sources: MatrixOfFactorSources,
+        profile: Profile,
+        account_addresses: IndexSet<AccountAddress>,
+        interactors: Arc<dyn KeysDerivationInteractors>,
+    ) -> Result<FactorInstancesProviderOutcome> {
+        Self::for_entity_mfa::<Account>(
+            cache,
+            matrix_of_factor_sources,
+            profile,
+            account_addresses,
+            interactors,
+        )
+        .await
+    }
+
+    /// Reads FactorInstances for every `factor_source` in matrix_of_factor_sources
+    /// on `network_id` of kind `account_mfa`,
+    /// meaning `(EntityKind::Account, KeyKind::TransactionSigning, KeySpace::Securified)`,
+    /// from cache, if any, otherwise derives more of that kind AND other kinds:
+    /// identity_veci, account_veci, identity_mfa
+    /// and saves into the cache and returns a collection of instances, per factor source,
+    /// split into factor instance to use directly and factor instances which was cached, into
+    /// the mutable `cache` parameter.
+    ///
+    /// We are always reading from the beginning of each FactorInstance collection in the cache,
+    /// and we are always appending to the end.
+    pub async fn for_persona_mfa(
+        cache: &mut FactorInstancesCache,
+        matrix_of_factor_sources: MatrixOfFactorSources,
+        profile: Profile,
+        persona_addresses: IndexSet<IdentityAddress>,
+        interactors: Arc<dyn KeysDerivationInteractors>,
+    ) -> Result<FactorInstancesProviderOutcome> {
+        Self::for_entity_mfa::<Persona>(
+            cache,
+            matrix_of_factor_sources,
+            profile,
+            persona_addresses,
+            interactors,
+        )
+        .await
+    }
+
+    /// Reads FactorInstances for every `factor_source` in matrix_of_factor_sources
+    /// on `network_id` of kind `account_mfa` or `identity_mfa` depending on Entity kind,
+    /// meaning `(EntityKind::_, KeyKind::TransactionSigning, KeySpace::Securified)`,
+    /// from cache, if any, otherwise derives more of that kind AND other kinds:
+    /// identity_veci, account_veci, identity_mfa/account_mfa
+    /// and saves into the cache and returns a collection of instances, per factor source,
+    /// split into factor instance to use directly and factor instances which was cached, into
+    /// the mutable `cache` parameter.
+    ///
+    /// We are always reading from the beginning of each FactorInstance collection in the cache,
+    /// and we are always appending to the end.
+    pub async fn for_entity_mfa<E: IsEntity>(
```

Comment:
is this intentionally public? Was expecting the usage of it to be directly for identity or account

Reply:
we dont know usage in Sargon yet. but I think we want to use this generic one actually, e.g. 

https://github.com/radixdlt/sargon-mfa/blob/factor_instances_provider/src/factor_instances_provider/provider/test_sargon_os.rs#L103-L112